x-pack/plugin/ql/src/test/java/org/elasticsearch/xpack/ql/expression/AttributeMapTests.java

```diff
@@ -61,6 +63,30 @@ public void testAttributeMapWithSameAliasesCanResolveAttributes() {
         assertTrue(newAttributeMap.get(param2.toAttribute()) == param2.child());
     }
 
+    public void testResolve() {
```

Comment:
I'd like to see more tests for dealing with:
a. same key, value association - (e,e)
b. handling of default value across association: resolveOrDefault(k, default) for a map with (k, default), (default, value) and also `resolveOrDefault(k, k)`
c. handling of cycles: (a, b), (b, c), (c,a ) with a mixture of default value in place.

Reply:
✓ Done.